File: backend/app/schemas/interview.py
from pydantic import BaseModel
from typing import List, Optional


class InterviewStartRequest(BaseModel):
    cv_id: str
    job_title: Optional[str] = None
    company: Optional[str] = None
    interview_type: str = "behavioral"  # behavioral, technical, case, cultural
    difficulty: int = 50  # 0-100
    session_length: int = 10  # 5, 10, 15
    language: str = "en"


class InterviewAnswerRequest(BaseModel):
    answer: str


class ChatMessage(BaseModel):
    id: int
    role: str  # system, ai-question, user, ai-feedback, next-prompt
    content: str
    question_number: Optional[int] = None
    question_type: Optional[str] = None
    scores: Optional[dict] = None
    strengths: Optional[List[str]] = None
    improvements: Optional[List[str]] = None
    model_answer: Optional[str] = None
    star_tip: Optional[bool] = None


class InterviewSession(BaseModel):
    id: str
    cv_id: str
    job_title: Optional[str] = None
    company: Optional[str] = None
    interview_type: str
    difficulty: int
    total_questions: int
    current_question: int = 0
    status: str = "active"  # active, ended
    messages: List[ChatMessage] = []
    created_at: str


class SessionSummary(BaseModel):
    id: str
    job_title: Optional[str] = None
    company: Optional[str] = None
    interview_type: str
    score: Optional[float] = None
    created_at: str


class SessionReport(BaseModel):
    session_id: str
    overall_score: float
    performance: dict  # radar chart data
    best_answer: Optional[str] = None
    areas_for_improvement: List[str] = []
    total_questions: int
    answered_questions: int

File: frontend/lib/api/interview.ts
import { request } from "./client";

export type InterviewType = "behavioral" | "technical" | "case" | "cultural";

export interface ChatMessage {
  id: number;
  role: "system" | "ai-question" | "user" | "ai-feedback" | "next-prompt";
  content: string;
  question_number?: number;
  question_type?: string;
  scores?: { overall: number; content: number; clarity: number };
  strengths?: string[];
  improvements?: string[];
  model_answer?: string;
  star_tip?: boolean;
}

export interface InterviewSession {
  id: string;
  cv_id: string;
  job_title?: string;
  company?: string;
  interview_type: InterviewType;
  difficulty: number;
  total_questions: number;
  current_question: number;
  status: "active" | "ended";
  messages: ChatMessage[];
  created_at: string;
}

export interface SessionSummary {
  id: string;
  job_title?: string;
  company?: string;
  interview_type: InterviewType;
  score?: number;
  created_at: string;
}

export interface SessionReport {
  session_id: string;
  overall_score: number;
  performance: Record<string, number>;
  best_answer?: string;
  areas_for_improvement: string[];
  total_questions: number;
  answered_questions: number;
}

export const interviewApi = {
  start: (data: {
    cv_id: string;
    job_title?: string;
    company?: string;
    interview_type?: InterviewType;
    difficulty?: number;
    session_length?: number;
    language?: string;
  }) =>
    request<InterviewSession>("/interview/start", {
      method: "POST",
      body: JSON.stringify(data),
    }),

  answer: (sessionId: string, answer: string) =>
    request<ChatMessage>(`/interview/${sessionId}/answer`, {
      method: "POST",
      body: JSON.stringify({ answer }),
    }),

  end: (sessionId: string) =>
    request<SessionReport>(`/interview/${sessionId}/end`, { method: "POST" }),

  listSessions: () => request<SessionSummary[]>("/interview/sessions"),

  getReport: (sessionId: string) =>
    request<SessionReport>(`/interview/${sessionId}/report`),
};
